src/app/directives/search.directive.ts

```diff
@@ -0,0 +1,43 @@
+import {Directive, ElementRef, HostBinding, Input} from '@angular/core';
+
+@Directive({
+  selector: '[search]',
+  exportAs: 'search'
+})
+export class SearchDirective {
+
+  @Input() set search(searchText: string) {
+    this.searchText = searchText;
+  }
+
+  @Input("itemId") id: string = "";
```

Comment:
usually this way is not convinient for developers. Let's use naming for decalring input name @Input() itemId: string = "";